### PR TITLE
Adding support for saving cloud resource data to context

### PR DESCRIPTION
**Pull request by mikejrizzo**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
N/A

## Description
This contribution enhances the `redlock-get-alert-details` command by adding a new argument: `resource_keys`.  With this new argument, specific keys from the resource data representing the cloud asset the alert is about can be saved to the context output generated by the command.

For example, a playbook author may wish to take different actions depending on arbitrary tag values set on a cloud asset.  In the current version, this tag data is shown in the printed output from the `redlock-get-alert-details` command, but it is not saved to Context, where it could be easily referenced by a playbook task.

The new `resource_keys` argument requires a specific list of key names to be provided.  This prevents unnecessary additional data from the resource data JSON being saved to the context, and only adds what is needed.  Multiple key names can be included as a comma separated list.

## Screenshots
This screenshot shows the resource config for a cloud asset which Prisma Cloud has raised an alert on.  Note that it contains a `tags` key in the JSON.
image

This screenshot shows the printed output from the command `!redlock-get-alert-details alertid=P1496074`, and shows that the `tags` key is returned.
image

This screenshot shows that in the current published version (2.0.17) that a limited set of resource data is stored to the context:
image

In the new version included in this pull request, including the argument `resource_keys="tags"` on the command creates a context entry that now contains the desired additional information from the resource JSON:
image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by omerKarkKatz — COMMENTED** [review]
Hey @mikejrizzo Great job the contribution!! 
Please look at my following suggestions and fix accordingly.
Also since this is an XSOAR supported pack please add a unit test checking that when executing the changed command with the parameter 'resource_keys' the key 'Data' is been added under the ec['Resources']
After that we can schedule a time for a demo.

File: Packs/PrismaCloud/Integrations/RedLock/RedLock.py [hunk 1]
```diff
@@ -213,28 +215,60 @@ def alert_to_context(alert):
...
+            'RiskDetail': {
+                'Rating': demisto.get(alert, 'riskDetail.rating'),
+                'Score': demisto.get(alert, 'riskDetail.riskScore.score')
+            },
+            'Resource': {
+                'ID': demisto.get(alert, 'resource.id'),
+                'Name': demisto.get(alert, 'resource.name'),
+                'Account': demisto.get(alert, 'resource.account'),
+                'AccountID': demisto.get(alert, 'resource.accountId')
+            }
```
[the remark this answers is not in the document]
Comment (mikejrizzo) [reply Packs/PrismaCloud/Integrations/RedLock/RedLock.py]: Done.

File: Packs/PrismaCloud/Integrations/RedLock/RedLock.yml [hunk 2]
```diff
@@ -399,6 +399,9 @@ script:
       - 'false'
       required: false
       secret: false
+    - description: List of additional keys to return from the resource JSON, specified as a comma separated list (e.g. "key1,key2,key3").  To preview all available resource JSON data, run redlock-get-alert-details with the "raw-response=true" option.
+      name: resource_keys
+      type: unknown
```
[the remark this answers is not in the document]
Comment (mikejrizzo) [reply Packs/PrismaCloud/Integrations/RedLock/RedLock.yml]: Done.

**Review by omerKarkKatz — COMMENTED** [review]
Modified the test description

File: Packs/PrismaCloud/Integrations/RedLock/RedLock_test.py [hunk 3]
```diff
@@ -398,3 +404,34 @@ def test_redlock_search_network(mocker):
...
+        Then
+        - Validate that the function returns the expected context JSON
+    Unit Test Scenario 2 - resource_keys argument included
+        Given
+        - Raw alert data
+        When
+        - resource_keys argument is set to 'tags'
+        Then
+        - Validate that the function returns the expected context JSON including the 'tags' element from the resource
+    """
```
Comment (omerKarkKatz) [inline Packs/PrismaCloud/Integrations/RedLock/RedLock_test.py:432]:
```suggestion
    """
        Given
            - Raw alert data
            - Raw alert data
        When
            - resource_keys argument is not set
            - resource_keys argument is set to 'tags'
        Then
            - Validate that the function returns the expected context JSON
            - Validate that the function returns the expected context JSON including the 'tags' element from the resource 
    """
```

**Review by omerKarkKatz — COMMENTED** [review]
Please change the test description section

File: Packs/PrismaCloud/Integrations/RedLock/RedLock_test.py [hunk 4]
```diff
@@ -398,3 +404,29 @@ def test_redlock_search_network(mocker):
...
+        Given
+            - Raw alert data
+            - Raw alert data
+        When
+            - resource_keys argument is not set
+            - resource_keys argument is set to 'tags'
+        Then
+            - Validate that the function returns the expected context JSON
+            - Validate that the function returns the expected context JSON including the 'tags' element from the resource
```
Comment (omerKarkKatz) [inline Packs/PrismaCloud/Integrations/RedLock/RedLock_test.py:426]:
```suggestion
            - Validate that the function returns the expected context JSON including the 'tags' element from the resource
```

**Review by omerKarkKatz — COMMENTED** [review]
removed trailing whitespace

**Review by omerKarkKatz — APPROVED** [review]

**Pull request merged**